docs/source/python/compute.rst

```diff
@@ -370,3 +370,165 @@ our ``even_filter`` with a ``pc.field("nums") > 5`` filter:
+   ...                         function_docs,
+   ...                             input_types,
+   ...                             output_type)
+   >>> 
+   >>> pc.call_function(function_name, [pa.scalar(10.1), pa.scalar(10.2), pa.scalar(20.2)])
+   <pyarrow.DoubleScalar: 123.22>
+
+Note that here the the final output is returned as an array. Depending the usage of vivid libraries
+inside the UDF, make sure it is generalized to support the passed input values and return suitable
+values. 
+
+Working with Datasets
+---------------------
+
+More generally, user-defined functions are usable everywhere a compute function
+can be referred to by its name. For example, they can be called on a dataset's
+column using :meth:`Expression._call`:
+
+Consider an instance where the data is in a table and you need to create a new 
+column using existing values in a column by using a mathematical formula.
+For instance, let's consider a simple affine operation on values using the
+mathematical expression, `y = mx + c`. We will be re-using the registered `affine`
+function.
+
+.. code-block:: python
+
+   >>> import pyarrow.dataset as ds
+   >>> sample_data = {'category': ['A', 'B', 'C', 'D'], 'value': [10.21, 20.12, 45.32, 15.12]}
+   >>> data_table = pa.Table.from_pydict(sample_data)
+   >>> dataset = ds.dataset(data_table)
+   >>> func_args = [pc.scalar(5.2), ds.field("value"), pc.scalar(2.1)]
+   >>> dataset.to_table(
+   ...             columns={
+   ...                 'projected_value': ds.field('')._call("affine", func_args),
+   ...                 'value': ds.field('value'),
+   ...                 'category': ds.field('category')
+   ...             })
+   pyarrow.Table
+   total_amount_projected($): int64
+   total_amount($): int64
+   trip_name: string
+   ----
+   total_amount_projected($): [[52,102,227,77]]
+   total_amount($): [[10,20,45,15]]
+   trip_name: [["A","B","C","D"]]
+
+Here note that the `ds.field('')_call()` returns an expression. The passed arguments
+to this function call are expressions not scalar values 
+(i.e `pc.scalar(5.2), ds.field("value"), pc.scalar(2.1)`). This expression is evaluated
+when the project operator uses this expression.
+
+Support
+-------
+
+It is defined that the current support is only for scalar functions. 
+A scalar function (:class:`arrow::compute::ScalarFunction`) executes elementwise operations
+on arrays or scalars. Generally, the result of such an execution doesn't
+depend on the order of values.
+
+There is a limitation in the support to UDFs in the current API.
+For instance, with project node, if a UDF is used as the compute function,
+it expects the function to be a scalar function. Although, this doesn't stop the user
+registering a non-scalar function and using it in a programme. 
+But it could lead to unexpected behaviors or errors when it is applied in such occasions. 
+The current UDF support could enhance with the addition of more settings to the API (i.e aggregate UDFs).
```

Comment:
```suggestion
Projection Expressions
^^^^^^^^^^^^^^^^^^^^^^

In the above example we used an expression to add a new column (`total_amount_projected`)
to our table.  Adding new, dynamically computed, columns to a table is known as "projection"
and there are limitations on what kinds of functions can be used in projection expressions.

A projection function must emit a single output value for each input row.  That output value
should be calculated entirely from the input row and should not depend on any other row.
For example, the "affine" function that we've been using as an example above is a valid
function to use in a projection.  A "cumulative sum" function would not be a valid function
since the result of each input rows depends on the rows that came before.  A "drop nulls"
function would also be invalid because it doesn't emit a value for some rows.
```